`no-unused-vars` error when using `eslint-plugin-mdx` without `eslint-plugin-react`.

### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

Latest

### Link to runnable example

https://stackblitz.com/edit/node-cppxmm?file=index.mdx

### Steps to reproduce

1. Open reproducible link
2. run `npx eslint ./index.mdx`


### Expected behavior

No error reported.

### Actual behavior


```
/home/projects/node-cppxmm/index.mdx
  1:8  error  'MyForm' is defined but never used  no-unused-vars
```

Enabling `eslint-plugin-react` would fix the error.

### Runtime

_No response_

### Package manager

_No response_

### OS

_No response_

### Build and bundle tools

_No response_

--- a/.changeset/twelve-pots-push.md
+++ b/.changeset/twelve-pots-push.md
@@ -1,0 +1,5 @@
+---
+"eslint-plugin-mdx": patch
+---
+
+Adds ecmaFeatures: { jsx: true } to premade configs

--- a/packages/eslint-plugin-mdx/src/configs/base.ts
+++ b/packages/eslint-plugin-mdx/src/configs/base.ts
@@ -5,6 +5,9 @@ export const base: Linter.LegacyConfig = {
   parserOptions: {
     sourceType: 'module',
     ecmaVersion: 'latest',
+    ecmaFeatures: {
+      jsx: true,
+    },
   },
   plugins: ['mdx'],
   processor: 'mdx/remark',

--- a/packages/eslint-plugin-mdx/src/configs/flat.ts
+++ b/packages/eslint-plugin-mdx/src/configs/flat.ts
@@ -11,6 +11,11 @@ export const flat: Linter.FlatConfig = {
   files: ['**/*.{md,mdx}'],
   languageOptions: {
     parser: eslintMdx,
+    parserOptions: {
+      ecmaFeatures: {
+        jsx: true,
+      },
+    },
     globals: {
       React: false,
     },

--- a/test/fixtures.test.ts
+++ b/test/fixtures.test.ts
@@ -2,6 +2,7 @@ import path from 'node:path'
 
 import eslintJs from '@eslint/js'
 import { TSESLint } from '@typescript-eslint/utils'
+import { Linter } from 'eslint'
 import prettierRecommended from 'eslint-plugin-prettier/recommended'
 import react from 'eslint-plugin-react'
 import globals from 'globals'
@@ -11,6 +12,7 @@ import { config, configs } from 'typescript-eslint'
 import * as mdx from 'eslint-plugin-mdx'
 
 const fixturesDir = path.resolve('test/fixtures')
+const eslintMajor = Number(new Linter().version.split('.')[0])
 
 const getCli = (lintCodeBlocks = false, fix?: boolean) => {
   const remarkConfigPath = path.resolve(
@@ -34,6 +36,11 @@ const getCli = (lintCodeBlocks = false, fix?: boolean) => {
           lintCodeBlocks,
         }),
         languageOptions: {
+          parserOptions: {
+            ecmaFeatures: {
+              jsx: true,
+            },
+          },
           globals: globals.node,
         },
         settings: {
@@ -52,7 +59,6 @@ const getCli = (lintCodeBlocks = false, fix?: boolean) => {
           'prefer-const': 'error',
           'react/jsx-curly-brace-presence': 'error',
           'react/self-closing-comp': 'error',
-          'no-unused-vars': 'error',
         },
       },
       {
@@ -112,10 +118,25 @@ describe('fixtures', () => {
     ONE_MINUTE,
   )
 
-  it(
+  /* eslint-disable jest/no-standalone-expect */
+  const itOnESLint10 = eslintMajor >= 10 ? it : it.skip
+  itOnESLint10(
     'should not report no-unused-vars for JSX component imports in MDX',
     async () => {
-      const cli = getCli()
+      // Uses only the plugin's built-in flat config without any user-land
+      // ecmaFeatures.jsx override, to verify the plugin sets it by default
+      const cli = new TSESLint.ESLint({
+        overrideConfigFile: true,
+        overrideConfig: [
+          eslintJs.configs.recommended,
+          mdx.configs.flat,
+          mdx.configs.flatCodeBlocks,
+          {
+            files: ['**/*.{md,mdx}'],
+            rules: { 'no-unused-vars': 'error' },
+          },
+        ],
+      })
 
       const results = await cli.lintFiles(
         path.join(fixturesDir, 'jsx-imports.mdx'),
@@ -130,6 +151,7 @@ describe('fixtures', () => {
     },
     ONE_MINUTE,
   )
+  /* eslint-enable */
 
   describe('lint code blocks', () => {
     it(

--- a/test/fixtures.test.ts
+++ b/test/fixtures.test.ts
@@ -12,7 +12,7 @@ import { config, configs } from 'typescript-eslint'
 import * as mdx from 'eslint-plugin-mdx'
 
 const fixturesDir = path.resolve('test/fixtures')
-const eslintMajor = Number(new Linter().version.split('.')[0])
+const isEslint10 = Number.parseInt(Linter.version, 10) >= 10
 
 const getCli = (lintCodeBlocks = false, fix?: boolean) => {
   const remarkConfigPath = path.resolve(
@@ -118,40 +118,32 @@ describe('fixtures', () => {
     ONE_MINUTE,
   )
 
-  /* eslint-disable jest/no-standalone-expect */
-  const itOnESLint10 = eslintMajor >= 10 ? it : it.skip
-  itOnESLint10(
-    'should not report no-unused-vars for JSX component imports in MDX',
-    async () => {
-      // Uses only the plugin's built-in flat config without any user-land
-      // ecmaFeatures.jsx override, to verify the plugin sets it by default
-      const cli = new TSESLint.ESLint({
-        overrideConfigFile: true,
-        overrideConfig: [
-          eslintJs.configs.recommended,
-          mdx.configs.flat,
-          mdx.configs.flatCodeBlocks,
-          {
-            files: ['**/*.{md,mdx}'],
-            rules: { 'no-unused-vars': 'error' },
-          },
-        ],
-      })
+  const test = isEslint10 ? it : it.skip
 
-      const results = await cli.lintFiles(
-        path.join(fixturesDir, 'jsx-imports.mdx'),
-      )
+  test('should not report no-unused-vars for JSX component imports in MDX', async () => {
+    // Uses only the plugin's built-in flat config without any user-land
+    // ecmaFeatures.jsx override, to verify the plugin sets it by default
+    const cli = new TSESLint.ESLint({
+      overrideConfigFile: true,
+      overrideConfig: [
+        mdx.configs.flat,
+        mdx.configs.flatCodeBlocks,
+        {
+          files: ['**/*.{md,mdx}'],
+          rules: { 'no-unused-vars': 'error' },
+        },
+      ],
+    })
 
-      const messages = results.flatMap(r => r.messages)
-      const unusedVarsMessages = messages.filter(
-        m => m.ruleId === 'no-unused-vars',
-      )
+    const results = await cli.lintFiles(
+      path.join(fixturesDir, 'jsx-imports.mdx'),
+    )
 
-      expect(unusedVarsMessages).toEqual([])
-    },
-    ONE_MINUTE,
-  )
-  /* eslint-enable */
+    const messages = results.flatMap(r => r.messages)
+
+    // eslint-disable-next-line jest/no-standalone-expect
+    expect(messages).toEqual([])
+  })
 
   describe('lint code blocks', () => {
     it(

--- a/test/fixtures.test.ts
+++ b/test/fixtures.test.ts
@@ -34,11 +34,6 @@ const getCli = (lintCodeBlocks = false, fix?: boolean) => {
           lintCodeBlocks,
         }),
         languageOptions: {
-          parserOptions: {
-            ecmaFeatures: {
-              jsx: true,
-            },
-          },
           globals: globals.node,
         },
         settings: {
@@ -57,6 +52,7 @@ const getCli = (lintCodeBlocks = false, fix?: boolean) => {
           'prefer-const': 'error',
           'react/jsx-curly-brace-presence': 'error',
           'react/self-closing-comp': 'error',
+          'no-unused-vars': 'error',
         },
       },
       {
@@ -112,6 +108,25 @@ describe('fixtures', () => {
           expect(output).toMatchSnapshot(relative(filePath))
         }
       }
+    },
+    ONE_MINUTE,
+  )
+
+  it(
+    'should not report no-unused-vars for JSX component imports in MDX',
+    async () => {
+      const cli = getCli()
+
+      const results = await cli.lintFiles(
+        path.join(fixturesDir, 'jsx-imports.mdx'),
+      )
+
+      const messages = results.flatMap(r => r.messages)
+      const unusedVarsMessages = messages.filter(
+        m => m.ruleId === 'no-unused-vars',
+      )
+
+      expect(unusedVarsMessages).toEqual([])
     },
     ONE_MINUTE,
   )

--- a/test/fixtures/jsx-imports.mdx
+++ b/test/fixtures/jsx-imports.mdx
@@ -1,0 +1,8 @@
+import { Card } from './card'
+import { Button } from './button'
+
+# Hello
+
+<Card>
+  <Button>Click me</Button>
+</Card>

